Update modding intro (#237)
bsmg/wiki: wiki/modding/intro.md

# Modding Intro
_Learn how to get started writing your own PC Mods_

::: warning
This guide is for making mods for the **PC** version of Beat Saber!

Make sure your game is modded before trying to make a mod.  
See instructions for [modding Beat Saber on PC.](/pc-modding.md)
:::

## Getting Started
Beat Saber is made in Unity 2019.3 using C# with .NET framework 4.6  

> You will need to download the latest version of [Visual Studio Community](https://visualstudio.microsoft.com/).

Reviewer: Somewhere around here, perhaps put a warning that this introduction assumes you know basic to intermediate levels of C# and Unity

### Edit your mod's Manifest
Fill out the `manifest.json` file with your information.  
The `name` and `id` keys are used to identify your mod. The ID should match the ID used when uploading your mod to BeatMods.  

::: warning
Do **not** remove the dependency on BSIPA. As of BSIPA v4.1 this is required for your mod to load.
:::

## Compiling
Build your plugin with `Build -> Build Solution` or <kbd>CTRL + SHIFT + B</kbd>  
Your compiled DLL should automatically be copied to the `Plugins` folder in your Beat Saber directory! This will be done for both debug and release builds.

::: tip NOTE
When you are ready to release your mod, select the `Release` option to make a Release build of your mod.

Building in Release mode will generate a packaged `.zip` file ready to upload to BeatMods.
:::

## Testing your mod in-game
To test if your mod is loaded in-game, you will need to launch Beat Saber with the BSIPA Console enabled. Add `--verbose` as a launch argument and run the game.  
For more information on launch arguments, see [here](/modding/#launch-args).

When you launch the game, you should see BSIPA load your mod in the console window.

> ![Testing console screenshot](~@images/modding/testing-console.png "Testing console screenshot")

Reviewer: Finally, I'd like to propose a `Next steps` section which would pretty much just be a link dump that people can dive in for  further information.

This would contain things like:
- Mention the BSMG `#pc-mod-dev` channel for mod development help
- A link to dnSpy for Beat Saber decompilation
- A link to BSIPA Docs for the Config system
- A link to Harmony for intermediate developers interested in method patching
- Zenject for experienced developers who know of Dependency Injection, since Beat Saber *heavily* uses this library.
  - While you're at it, might as well mention SiraUtil too, since that allows developers to easily hook into Beat Saber's Dependency Injection system.

Reviewer 2: This looks good.

Reviewer 3: How does the next steps writeup look?

Reviewer: > For experienced developers, you may be interested learning about Zenject for Dependency Injection which the game heavily uses.

I think this could be worded better, something like `For experienced developers, you may be interested in learning about Zenject, the Dependency Injection system used heavily by Beat Saber.` The second sentence is fine.

My only other nitpick is that you should probably add a link to [Harmony](https://github.com/pardeike/Harmony) for patching the games code

Reviewer 3: Late night me skipped the harmony bullet point... It has been added

Reviewer: Correct, Harmony will come bundled with BSIPA

::: warning
Do **not** remove the dependency on BSIPA. As of BSIPA v4.1 this is required for your mod to load.
:::

## Compiling
Build your plugin with `Build -> Build Solution` or <kbd>CTRL + SHIFT + B</kbd>  
Your compiled DLL should automatically be copied to the `Plugins` folder in your Beat Saber directory! This will be done for both debug and release builds.

::: tip NOTE
When you are ready to release your mod, select the `Release` option to make a Release build of your mod.

Building in Release mode will generate a packaged `.zip` file ready to upload to BeatMods.
:::

## Testing your mod in-game
To test if your mod is loaded in-game, you will need to launch Beat Saber with the BSIPA Console enabled. Add `--verbose` as a launch argument and run the game.  
For more information on launch arguments, see [here](/modding/#launch-args).

When you launch the game, you should see BSIPA load your mod in the console window.

![Testing console screenshot](~@images/modding/testing-console.png "Testing console screenshot")

## Next Steps
Here are some useful resources in continuing your modding career.

> * If you need help with developing mods, you can ask in [BSMG's #pc-mod-dev](https://discord.gg/beatsabermods) channel.

Reviewer 4: ```suggestion
* If you need help with developing mods, you can ask in `#pc-mod-dev` on the [BSMG Discord](https://discord.gg/beatsabermods).
```

> * If you want to decompile code, checkout [dnSpy](https://github.com/dnSpy/dnSpy/releases)

Reviewer 4: ```suggestion
* If you want to decompile code, check out [dnSpy](https://github.com/dnSpy/dnSpy/releases)
```
